---
title: "Angsd and XP-CLR"
author: "Bo Peng"
date: "2025-05-04"
output:
  html_document:
    df_print: paged
  word_document: default
---

#### Calculate Theta Values

```{bash}
#!/bin/bash -l
#SBATCH -D /work/agro932/bpeng4/PopGen_HW/
#SBATCH -o /work/agro932/bpeng4/PopGen_HW/largedata/slurm-log/stdout-%A_%a.txt
#SBATCH -e /work/agro932/bpeng4/PopGen_HW/largedata/slurm-log/stderr-%A_%a.txt
#SBATCH -J SNP_calling
#SBATCH -t 168:00:00
#SBATCH --array=56,58,63,68,69,86,87,88,89,90  # Define an array job with specific task IDs
#SBATCH --mem-per-cpu=5G       
#SBATCH --partition=benson,batch
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=36


set -e
set -u

- Step1: Index Your Reference Genome

module load samtools/1.20 angsd/0.937
cd /work/agro932/bpeng4/PopGen_HW/largedata
samtools faidx chr1.fasta


- Step 2: Prepare BAM File List

ls sorted*bam > bam_list.txt


- Step 3: Generate Site Frequency Spectrum (SFS)

angsd -bam bam_list.txt  -doMaf 1 -doMajorMinor 1 -uniqueOnly 1 -minMapQ 30 -minQ 20 -minInd 6 -doSaf 1 -anc chr1.fasta -GL 2 -out maize -P 4


- Step 4: Compute theta
# Step 4a: Estimate the SFS from SAF index
realSFS maize.saf.idx -P 4 > maize.sfs

# Step 4b: Compute theta using the SFS
realSFS saf2theta maize.saf.idx -outname maize -sfs maize.sfs -fold 1 -P 4


- Step 5: Extract theta estimates using a 5 kb window size with a 1 kb step size.

thetaStat do_stat maize.thetas.idx -win 5000 -step 1000 -outnames maize.thetasWindow5kb_1kb
cp maize.thetasWindow5kb_1kb.pestPG /work/agro932/bpeng4/PopGen_HW/cache/
```



### plot results

```{r,eval=F}
library("data.table")
d <- fread("/Users/bopeng/Documents/GitHub/PopGen_HW/cache/maize.thetasWindow5kb_1kb.pestPG", header=T, data.table=F)

pi=d$tP/d$nSites
plot(d$tW/d$nSites,d$tP/d$nSites,xlab="Watterson Theta",ylab="Pairwise Theta",pch=16,col=adjustcolor("blue",alpha.f = 0.5))
plot(d$WinCenter/1000, d$tP/d$nSites, xlab = "Chr1 (kp)", ylab="Pairwise Theta")
```

```{r}
plot(d$WinCenter/1000, d$Tajima/d$nSites, xlab = "Chr1 (kp)", ylab="Tajima's D")
```

### an example about how to interpret the results in text

#### 1. First, provide a brief description of the analysis, but include details such as data, software, and parameters in the 'Materials and Methods' section.

In this simulated experiment, we generated 10 individuals from two populations (see M&M for details).

#### 2. Begin by describing the results, and if necessary, provide a brief reminder or explanation to help educate the reader.

Tajima's D, a measure of the site frequency spectrum (SFS) distribution between two populations, is expected to be around 0 if a genomic region is neutral (cite). As shown in the figure, most of the sites have values close to 0, consistent with our simulation, which did not include any sites under selection.

#### 3. Highlight any exciting discoveries.

However, we observed that in the 10 kb region, the values of Tajima's D are slightly above 0, with an average of approximately 0.03. A positive Tajima's D can indicate balancing selection, as it suggests higher-than-expected nucleotide diversity. This typically occurs because both alleles have been maintained at the balanced sites in the population, allowing more mutations to accumulate. 

#### 4. Offer alternative explanations, can be backed up by other's studies

However, this slight elevation above 0 is more likely caused by a high mutation rate in the simulation, consistent with previous studies (citation).



