---
title: "HW2"
output: html_document
date: "2025-03-10"
name: Bo Peng

output:
  html_document:
    df_print: paged
  word_document: default
---


# Reference genome
cd 

#### Change to `largedata` folder:

```{bash, eval=FALSE}
cd /work/agro932/bpeng4/PopGen_HW/largedata
```

#### Then use `wget` to download the reference genome:

```{bash, eval=FALSE}
wget https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/release-60/fasta/zea_mays/dna/Zea_mays.Zm-B73-REFERENCE-NAM-5.0.dna.chromosome.1.fa.gz

### then unzip it
gunzip Zea_mays.Zm-B73-REFERENCE-NAM-5.0.dna.chromosome.1.fa.gz

### then check the file with less
less Zea_mays.Zm-B73-REFERENCE-NAM-5.0.dna.chromosome.1.fa | grep ">"

# extracts the first 100k from chr1 of the genome 
module load seqtk

seqtk subseq Zea_mays.Zm-B73-REFERENCE-NAM-5.0.dna.chromosome.1.fa <(echo -e "1\t0\t100000") > chr1.fasta

```

`<(echo -e "1\t0\t100000")`: Specifies the chromosome (1) and range (1-100000).


#### next step, let's get the GFF file

```{bash, eval=FALSE}
wget https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/release-60/gff3/zea_mays/Zea_mays.Zm-B73-REFERENCE-NAM-5.0.60.chromosome.1.gff3.gz

### then unzip it
gunzip Zea_mays.Zm-B73-REFERENCE-NAM-5.0.60.chromosome.1.gff3.gz

### then check the file with less
less Zea_mays.Zm-B73-REFERENCE-NAM-5.0.60.chromosome.1.gff3

```

--


#### check how many reads

```{bash, eval=FALSE}
wc -l SRR11119156_1.fastq
```

#### Align the sequencing by BWA

```{bash, eval=FALSE}
#!/bin/bash -l
#SBATCH -D /work/agro932/bpeng4/PopGen_HW
#SBATCH -o /work/agro932/bpeng4/PopGen_HW/slurm-script/align-stdout-%j.txt
#SBATCH -e /work/agro932/bpeng4/PopGen_HW/slurm-script/align-stderr-%j.txt
#SBATCH -J Bwa_align
#SBATCH -t 24:00:00
#SBATCH --mail-user=bpeng4@huskers.unl.edu
#SBATCH --mail-type=END #email if ends
#SBATCH --mail-type=FAIL #email if fails
set -e
set -u

module load bwa samtools
cd /work/agro932/bpeng4/PopGen_HW/largedata

# Align reads to the reference genome
for i in 56 58 63 68 69 86 87 88 89 90; do
bwa mem chr1.fasta SRR111191${i}_1.fastq SRR111191${i}_2.fastq | samtools view -bSh - > SRR111191${i}.bam
done

# Sort BAM files
for i in *.bam; do
samtools sort "$i" -o "sorted_$i"
done

# Index the sorted BAM files
for i in sorted_*.bam; do
samtools index "$i"
done
```

#### Check mapping statistics

```{bash, eval=FALSE}
samtools flagstat sorted_l1.bam
```






