---
title: "HW2"
output: html_document
date: "2025-03-10"
name: Bo Peng

output:
  html_document:
    df_print: paged
  word_document: default
---

### 2. Obtain the SNP calls  with `samtools`

```{bash}
#!/bin/bash -l
#SBATCH -D /work/agro932/bpeng4/PopGen_HW/
#SBATCH -o /work/agro932/bpeng4/PopGen_HW/largedata/slurm-log/stdout-%A_%a.txt
#SBATCH -e /work/agro932/bpeng4/PopGen_HW/largedata/slurm-log/stderr-%A_%a.txt
#SBATCH -J SNP_calling
#SBATCH -t 168:00:00
#SBATCH --array=56,58,63,68,69,86,87,88,89,90  # Define an array job with specific task IDs
#SBATCH --mem-per-cpu=5G       
#SBATCH --partition=benson,batch
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=36


set -e
set -u

# Load required modules
module load samtools bcftools

# Change to the working directory
cd largedata

# Define sample ID from the SLURM array index
SAMPLE_ID=$SLURM_ARRAY_TASK_ID

# Step 1: Index the sorted BAM file (if not already indexed)
if [ ! -f sorted_SRR111191${SAMPLE_ID}.bam.bai ]; then
    samtools index sorted_SRR111191${SAMPLE_ID}.bam
fi

# Step 2: Generate mpileup file
bcftools mpileup -f chr1.fasta sorted_SRR111191${SAMPLE_ID}.bam > sorted_SRR111191${SAMPLE_ID}.mpileup

# Step 3: Call SNPs
bcftools call -m --ploidy 1 -Ob -o SRR111191${SAMPLE_ID}.bcf SRR111191${SAMPLE_ID}.mpileup

# Step 4: Index the VCF file
bcftools index SRR111191${SAMPLE_ID}.bcf

```

submit job using the below code:

```{bash, eval=FALSE}
sbatch --licenses=common slurm-script/snpcalling.sh
```


# Merging All VCF Files (After All Jobs Complete)

```{bash}
module load bcftools
bcftools merge -Ob -o merged.bcf SRR111191${56,58,63,68,69,86,87,88,89,90}.bcf
bcftools index merged.bcf

# Filter for Biallelic SNPs
bcftools view -m2 -M2 -v snps -Ob -o merged_biallelic_snps.bcf merged.bcf


# get a summary of the VCF file
bcftools stats  merged_biallelic_snps.bcf > vcf_summary.txt
```


# Convert VCF to Tab-Delimited Format

```{bash}
bcftools query -f '%CHROM\t%POS\t%REF\t%ALT\t%QUAL\t%DP[\t%GT]\n' merged_biallelic_snps.bcf > snp_calls.txt
```
Chromosome (CHROM)
Position (POS)
Reference allele (REF)
Alternate allele (ALT)
Quality (QUAL)
Depth of coverage (DP)
Genotypes (GT) for each sample

# Convert VCF to PLINK format

```{bash}
module laod plink/1.90
plink --bcf merged_biallelic_snps.bcf --allow-extra-chr --make-bed --out merged_variants

# move merged files to a new folder
mkdir merged
mve merged_varaints* merged/

# get the missing data and maf PCA
plink --bfile merged_variants --allow-extra-chr --missing --freq --pca 20 --out merged_variants

```





